### ISSUE TITLE
E6 Modules named import variable "types" treated as non-existent

### ISSUE THREAD

**Issue by petterm** [SonarSource/SonarJS#995]
**SonarJS version:** 4.1 (build 6085)
**SonarQube version:** Version 7.1 (build 11001)<br>

**Rule key:** javascript:S3827

**Reproducer** 
```
import { types } from 'some/path/here';

export default (state = {}, action) => {
    switch (action.type) {
    case types.SOMETHING:
        return { test: true };
    default:
        return state;
    }
};
```

**Result**
Line 5: "types" does not exist. Change its name or declare it so that its usage doesn't result in a "ReferenceError".

**Expected behavior**
No errors at all.

If i rename the import like the example below Sonarqube finds no errors.
```
import { types as someTypes } from 'some/path/here';
```

**Comment by petterm** [comment]
Probably related but I just noted a similar false positive for rule `S1128`.

```
import { types } from 'some/path/here';
             ^ Remove this unused import of 's'.
```

**Comment by thomasvsundert** [comment]
We are experiencing the exact same issue, which -unfortunately- makes SonarJS pretty much unusable in our applications.

**Comment by vilchik-elena** [comment]
@petterm @thomasvsundert thanks for raising the point! Indeed annoying behaviour.

While it's not fixed, as a workaround I can suggest configuring `sonar.javascript.globals` (add `types` to the list, so that it's considered existing for `S3827`).

And you could disable `S1128` (I know it's kind of pain, as rule is in default profile) to avoid `Remove this unused import of 's'.`.

**Comment by thomasvsundert** [comment]
Most of our projects have over 300 "non-existing" variables, so adding them to the configuration is quite tedious :)
